src/java.base/share/classes/sun/net/www/protocol/http/DigestAuthentication.java

```diff
@@ -60,19 +64,39 @@
 
     private String authMethod;
 
+    private static final String secPropName =
+        "jdk.httpdigest.defaultDisabledAlgorithms";
+
     private static final String compatPropName = "http.auth.digest." +
         "quoteParameters";
 
     // default set of disabled message digest algorithms that may not be
     // used for proxy connections, or plain text http server connections
 
-    // This will probably be expanded to include SHA-1 eventually
-    private static final Set<String> defDisabledAlgs =
-        Set.of("MD5");
+    private static final Set<String> defDisabledAlgs = getDefaultAlgs();
+
+    private static Set<String> getDefaultAlgs() {
+        Set<String> disabledAlgs = new HashSet<>();
+        @SuppressWarnings("removal")
+        String secprops = AccessController.doPrivileged(
+            new PrivilegedAction<>() {
```

Comment:
could use a lambda instead of an anonymous class?

Reply:
Had tried it first and compiler didn't know whether lambda is a PrivilegedAction or a PrivilegedExceptionAction, but it seems a cast works. Will change it.

Reply:
or declare a local variable that holds the lambda - that would avoid the cast.